portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerImpl.java

```diff
@@ -0,0 +1,82 @@
+package org.dataportabilityproject.worker;
+
+import com.google.common.base.Preconditions;
+import com.google.inject.Inject;
+import java.io.IOException;
+import org.dataportabilityproject.PortabilityCopier;
+import org.dataportabilityproject.ServiceProviderRegistry;
+import org.dataportabilityproject.cloud.interfaces.CloudFactory;
+import org.dataportabilityproject.job.JobDao;
+import org.dataportabilityproject.job.PortabilityJob;
+import org.dataportabilityproject.shared.PortableDataType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class WorkerImpl {
+  private final Logger logger = LoggerFactory.getLogger(WorkerImpl.class);
+  private final CloudFactory cloudFactory;
+  private final JobPollingService jobPollingService;
+  private final JobDao jobDao;
+  private final ServiceProviderRegistry registry;
+
+  @Inject
+  WorkerImpl(
+      CloudFactory cloudFactory,
+      JobDao jobDao,
+      JobPollingService jobPollingService,
+      ServiceProviderRegistry registry) {
+    this.cloudFactory = cloudFactory;
+    this.jobPollingService = jobPollingService;
+    this.jobDao = jobDao;
+    this.registry = registry;
+  }
+
+  void processJob() {
+    // Start the polling service to poll for an unassigned job and when it's ready.
+    pollForJob();
+
+    // Start the processing
+    PortabilityJob job = getJob(jobDao);
+
+    // Only load the two providers that are doing actually work.
+    // TODO(willard): Only load two needed services here, after converting service name to class
+    // name in the DAO.
+
+    processJob(job);
+    logger.info("Successfully processed jobId: {}", WorkerJobMetadata.getInstance().getJobId());
```

Comment:
In a follow up, we can move WorkerJobMetadata to a @Singleton class

Reply:
Ba Humbug, why wait till a new CL, I added that change to this one.